src/System.Management.Automation/namespaces/LocationGlobber.cs

```diff
@@ -3339,8 +3339,10 @@ internal static string GetDriveQualifiedPath(string path, PSDriveInfo drive)
                 }
                 else
                 {
-                    if (path.StartsWith(StringLiterals.DefaultPathSeparatorString, StringComparison.Ordinal) || 
-                        path.StartsWith(StringLiterals.AlternatePathSeparatorString, StringComparison.Ordinal)) 
+                    if ((path.StartsWith(StringLiterals.DefaultPathSeparatorString, StringComparison.Ordinal) || 
+                         path.StartsWith(StringLiterals.AlternatePathSeparatorString, StringComparison.Ordinal)) &&
+                        !(path.Equals(StringLiterals.DefaultPathSeparatorString, StringComparison.Ordinal) ||
+                         path.Equals(StringLiterals.AlternatePathSeparatorString, StringComparison.Ordinal)))
```

Comment:
These comparison is tedious. We can achieve the same with the following:
```
if (path.IndexOfAny(new char[]{ '\\', '/' }, 0, 1) == 0 && path.Length > 1)
{
    treatAsRelative = false;
}
```
Note that since this code path may be hot, the `new char[]{ '\\', '/' }` probably should be declared as a static char array in `StringLiterals` using `StringLiterals.DefaultPathSeparator` and `StringLiterals.AlternatePathSeparator`. This is based on [Performance considerations](https://github.com/PowerShell/PowerShell/blob/master/docs/dev-process/coding-guidelines.md#performance-considerations).